src/NHibernate.Test/TypesTest/CharClassFixture.cs

```diff
@@ -31,5 +32,33 @@ public void ReadWrite()
 				s.Flush();
 			}
 		}
+
+		[Test]
+		public void ParameterTypeForAnsiCharInLinq()
+		{
+			using (var logSpy = new SqlLogSpy())
+			using (var session = OpenSession())
+			{
+				var result = (from e in session.Query<CharClass>()
+							where e.AnsiChar == 'B'
+							select e).ToList();
+				
+				Assert.That(logSpy.GetWholeLog(), Does.Contain("Type: AnsiString"));
+			}
+		}
+
+		[Test]
+		public void ParameterTypeForCharInAnsiStringInLinq()
```

Comment:
Added test case that I described in https://github.com/nhibernate/nhibernate-core/issues/2963#issuecomment-1001112687

Reply:
Hm... It also fails in 5.2 for some other reasons (I didn't look into it). So not really  a regression - so let's not touch it in 5.3...

Reply:
It doesn't work in hql because `AnsiString[0]` is translated to function `substring` which is registered by `AnsiSubstringFunction` that has `NHibernateUtil.String` as the return type. With `5.3` it is detected as `StringFixedLengthSqlType`, which is better than before but still not correct, as it should be `AnsiStringFixedLengthSqlType`.